### DISCUSSION TITLE
Which model file should i use while deploying custom model that i created using monai label using monai-deploy-app-sdk.

### DISCUSSION THREAD

**Discussion by ArpanGyawali** [Project-MONAI/MONAILabel#1515] [General]
My task: Skull stripping
What I did:
- Rewrote the config, infers, and trainers file of the segmentation-spleen model within radiology by changing labels, roi, and transforms as required and renamed it as segmentation-brain.
- Then, I annotated my own dataset for skull stripping using 3D slicer. The only label used was "brain"
- Then I trained the images and its label using the custom segmentation-brain model using monai label extension in 3d slicer and got a dice score of about 0.98 and I am happy with the result.

Now I am trying to deploy the model for inference using monai-deploy-app-sdk. 
- For that, I used the model.pt file saved inside the `apps/radiology/models/segmentation brain` directory in the command
`python app.py -i input_dir -o output_dir -m model.pt`

But I am getting the error: `ItemNotExistsError: A predictor of the model is not set `
How can I get the saved inference model file that can be used while deploying? Do we need a torchscript file of the model? If yes then how to get that?

Any help would be very appreciated.

**Comment by diazandr3s** [comment]
Hi @ArpanGyawali,

Thanks for opening this discussion.
Do you want to deploy the model for batch inference? Have you tried using the MONAI Label server for inference or batch inference? You could serve the model with MONAI Label as well.

If you want to continue with MONAI deploy, I'd suggest opening a discussion in the MONAI Deploy repo. 

Or maybe @MMelQin or @gigony can comment on this as well.

**Reply by ArpanGyawali** [reply]
Thanks @diazandr3s for the reply.
I trained the model and I want to create an executable for others to use my model for inference of new images for skull stripping easily without having to open the Monai label in 3D slicer. After i could be able tp make the executable run in my local computer, my plan is to create a docker image out of it. 
But while following the documentation on monai deploy, using the model file I got from monai label, I got the error mentioned above.

What would be the best option for me to do?
I have tried reaching out in the monai deploy repo too. I am waiting for the reply. I would love to have a further discussion on this and get help.

**Comment by ArpanGyawali** [comment]
@diazandr3s @MMelQin @gigony

**Reply by MMelQin** [reply]
The MONAI Deploy App SDK model network loader supports TorchScript, so the model needs to be traced/scripted to a TorchScript file. Better yet, if a MONAI Bundle is created (assuming just with pre_processing, post_processing, and infer), the built-in MONAI Bundle Inference operator can be used to automate these steps in the app.

**Reply by diazandr3s** [reply]
You could also run MONAI Label as a standard python script. For this, run the main function: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/main.py#L264
Here are the arguments to pass: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/main.py#L289-L291

**Comment by wyli** [comment]
cc @ericspod for any bundle related comments/suggestions, pls see also https://github.com/Project-MONAI/model-zoo/discussions/490

**Reply by ericspod** [reply]
I've replied to @ArpanGyawali in that discussion. Thanks!
